### ISSUE TITLE
Add barong JWT integration

### ISSUE THREAD

**Issue by monsieurleberre** [trakx/trakx-tools#42]
To be able to have web pages for issue redeem of index, for rebalancing, etc, available only to market makers, admins, etc.
Permission can be checked with labels easily after that.
CF Openware barong npm package, just translate in c#!
https://www.npmjs.com/package/@openware/node-auth-barong
https://github.com/openware/node-auth-barong

This might help
https://www.youtube.com/watch?v=PFHK4YDJfaI

**Comment by monsieurleberre** [comment]
![image](https://user-images.githubusercontent.com/4638821/84657530-53494300-af0c-11ea-87ed-e0881be940e3.png)

**Comment by foraone** [comment]
# Changes to Envoy config
File to change:  `templates/config/gateway/envoy.yaml.erb`
1. Create a new cluster at the end of the file (demo):
```  - name: demo
    connect_timeout: 0.25s
    type: strict_dns
    lb_policy: round_robin
    hosts:
    - socket_address:
        address: demo
        port_value: 3000
```
2. Add route (closer to top in route_config):
```              - match:
                  prefix: "/api/v2/demo"
                route:
                  cluster: demo
```

In this demo we also edited file `template/compose/app.yaml` to add additional service: ```

**Comment by sergiisav** [comment]
@monsieurleberre any idea why I am tagged here?
@foraone full namesake huh? :D where you are from?

Seems like you are having an interesting project here! :) do you accept newcomers? 

p.s. sorry for the flood. you can drop a message on skype: pinedev

**Comment by monsieurleberre** [comment]
@sergiisav Hello! welcome here, yes we accept newcomers with great pleasure! It is an open repo, so feel free to ask if you think something can be improved 👍

**Comment by AymericNoel** [comment]
@foraone , In the video we see that the labels are returned but when we decrypt the tokens here, they are not returned, can you help please ?

**Comment by AymericNoel** [comment]
Resolved in #158
